Workers keep getting killed

Salut !

Super projet, j'avais fais un script de mon côté mais le tien est carrément plus complet !

Néanmoins j'ai un problème récurrent, soit mes workers se font timeout, soit SIGKILL. J'ai check j'ai largement assez de CPU et de RAM pour le faire fonctionner.
Au delà de ça, j'ai pas mal d'erreurs similaires à #4 alors que j'ai bien tout configuré.

**Ma conf**: 
```yaml
  yggratio-flaresolverr:
    # DockerHub mirror flaresolverr/flaresolverr:latest
    image: ghcr.io/flaresolverr/flaresolverr:latest
    container_name: yggratio-flaresolverr
    networks:
      yggratio-network:
    environment:
      - LOG_LEVEL=${LOG_LEVEL:-info}
      - LOG_HTML=${LOG_HTML:-false}
      - CAPTCHA_SOLVER=${CAPTCHA_SOLVER:-none}
      - TZ=Europe/Paris
    restart: always

  yggratio-yggproxy:
    image: ghcr.io/limedrive/ygg-rss-proxy:latest
    container_name: yggratio-yggproxy
    networks:
      yggratio-network:
    environment:
        YGG_USER: ${YGG_USER}
        YGG_PASS: ${YGG_PASS}
        RSS_HOST: yggratio-yggproxy
        RSS_PORT: 5000
        RSS_SHEMA: http
        FLARESOLVERR_HOST: yggratio-flaresolverr
        FLARESOLVERR_SHEMA: http
        FLARESOLVERR_PORT: 8191
        GUNICORN_PORT: 5000
        LOG_LEVEL: DEBUG
    restart: always
    depends_on:
        - yggratio-flaresolverr
```

**Exemples de logs:**

```log
2024-06-14T19:50:30.011665+0000 INFO ygg_basic_login Successfully authenticated to YGG
2024-06-14T19:50:30.012280+0000 DEBUG ygg_cloudflare_login Session cookies: <CookieJar[<Cookie cf_clearance=XXXX for .ygg.re/>, <Cookie ygg_=XXXXXX for .ygg.re/>]>
2024-06-14T19:50:30.016907+0000 DEBUG _new_conn Starting new HTTPS connection (1): www.ygg.re:443
2024-06-14T19:50:35.724742+0000 DEBUG _make_request https://www.ygg.re:443 "GET /rss?action=generate&type=subcat&id=2183&********
[2024-06-14 19:50:39 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:31)
[2024-06-14 19:50:40 +0000] [1] [ERROR] Worker (pid:31) was sent SIGKILL! Perhaps out of memory?
[2024-06-14 19:50:40 +0000] [35] [INFO] Booting worker with pid: 35
``` 

```
2024-06-14T19:18:31.638570+0000 DEBUG _make_request https://www.ygg.re:443 "GET /rss?action=generate&type=subcat&id=2184&********
[2024-06-14 19:18:36 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:12)
[2024-06-14 19:18:37 +0000] [1] [ERROR] Worker (pid:12) was sent SIGKILL! Perhaps out of memory?
[2024-06-14 19:18:37 +0000] [17] [INFO] Booting worker with pid: 17
2024-06-14T19:25:55.321712+0000 DEBUG _new_conn Starting new HTTPS connection (1): www.ygg.re:443
[2024-06-14 19:25:56 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:17)
[2024-06-14 19:25:57 +0000] [1] [ERROR] Worker (pid:17) was sent SIGKILL! Perhaps out of memory?
[2024-06-14 19:25:57 +0000] [18] [INFO] Booting worker with pid: 18
2024-06-14T19:25:58.414919+0000 DEBUG _new_conn Starting new HTTPS connection (1): www.ygg.re:443
2024-06-14T19:26:09.380631+0000 DEBUG _make_request https://www.ygg.re:443 "GET /auth/login HTTP/1.1" 403 None
2024-06-14T19:26:10.356895+0000 INFO ygg_login Cloudflare is enabled, using FlareSolverr
2024-06-14T19:26:11.173063+0000 DEBUG _new_conn Starting new HTTP connection (1): yggratio-flaresolverr:8191
2024-06-14T19:26:11.319947+0000 DEBUG _make_request http://yggratio-flaresolverr:8191 "GET / HTTP/1.1" 200 172
[2024-06-14 19:26:27 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:15)
[2024-06-14 19:26:28 +0000] [1] [ERROR] Worker (pid:15) was sent SIGKILL! Perhaps out of memory?
[2024-06-14 19:26:28 +0000] [19] [INFO] Booting worker with pid: 19
2024-06-14T19:26:32.999958+0000 DEBUG _new_conn Starting new HTTPS connection (1): www.ygg.re:443
2024-06-14T19:26:40.036645+0000 DEBUG _make_request https://www.ygg.re:443 "GET /auth/login HTTP/1.1" 403 None
2024-06-14T19:26:40.447775+0000 INFO ygg_login Cloudflare is enabled, using FlareSolverr
2024-06-14T19:26:40.451967+0000 DEBUG _new_conn Starting new HTTP connection (1): yggratio-flaresolverr:8191
2024-06-14T19:26:40.456729+0000 DEBUG _make_request http://yggratio-flaresolverr:8191 "GET / HTTP/1.1" 200 172
[2024-06-14 19:26:58 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:16)
[2024-06-14 19:26:59 +0000] [1] [ERROR] Worker (pid:16) was sent SIGKILL! Perhaps out of memory?
2024-06-14T19:26:59.736896+0000 DEBUG _new_conn Starting new HTTPS connection (1): www.ygg.re:443
[2024-06-14 19:27:00 +0000] [20] [INFO] Booting worker with pid: 20
```

```
[2024-06-14 19:16:31 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:9)
[2024-06-14 19:16:32 +0000] [1] [ERROR] Worker (pid:9) was sent SIGKILL! Perhaps out of memory?
[2024-06-14 19:16:32 +0000] [13] [INFO] Booting worker with pid: 13
2024-06-14T19:16:33.035337+0000 DEBUG _get_conn Resetting dropped connection: www.ygg.re
2024-06-14T19:16:33.480692+0000 DEBUG _make_request https://www.ygg.re:443 "GET /auth/login HTTP/1.1" 403 None
2024-06-14T19:16:33.482903+0000 INFO ygg_login Cloudflare is enabled, using FlareSolverr
2024-06-14T19:16:33.485263+0000 DEBUG _new_conn Starting new HTTP connection (1): yggratio-flaresolverr:8191
2024-06-14T19:16:33.490211+0000 DEBUG _make_request http://yggratio-flaresolverr:8191 "GET / HTTP/1.1" 200 172
[2024-06-14 19:17:02 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:8)
[2024-06-14 19:17:02 +0000] [8] [ERROR] Error handling request /rss?action=generate&type=subcat&id=2178&passkey=XXXXX
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/sync.py", line 178, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1498, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 878, in full_dispatch_request
    rv = self.preprocess_request()
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1253, in preprocess_request
    rv = self.ensure_sync(before_func)()
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/ygg_rss_proxy/app.py", line 34, in before_request
    init_session()
  File "/app/ygg_rss_proxy/session_manager.py", line 33, in init_session
    new_session()
  File "/app/ygg_rss_proxy/session_manager.py", line 15, in new_session
    ygg_session = ygg_login()
                  ^^^^^^^^^^^
  File "/app/ygg_rss_proxy/auth.py", line 150, in ygg_login
    return ygg_cloudflare_login(session, ygg_playload)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/ygg_rss_proxy/auth.py", line 74, in ygg_cloudflare_login
    response = fs_solver.request_get(url="https://www.ygg.re")
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/ygg_rss_proxy/fspy/solver.py", line 198, in request_get
    return self._do_the_work_for_get_post(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/ygg_rss_proxy/fspy/solver.py", line 157, in _do_the_work_for_get_post
    response = self.req_session.post(self.flare_solverr_url, json=payload)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 637, in post
    return self.request("POST", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 793, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 537, in _make_request
    response = conn.getresponse()
               ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 466, in getresponse
    httplib_response = super().getresponse()
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/http/client.py", line 1395, in getresponse
    response.begin()
  File "/usr/local/lib/python3.11/http/client.py", line 325, in begin
    version, status, reason = self._read_status()
                              ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/http/client.py", line 286, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/socket.py", line 706, in readinto
    return self._sock.recv_into(b)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/base.py", line 203, in handle_abort
    sys.exit(1)
SystemExit: 1
[2024-06-14 19:17:03 +0000] [8] [INFO] Worker exiting (pid: 8)
[2024-06-14 19:17:03 +0000] [1] [ERROR] Worker (pid:8) was sent SIGKILL! Perhaps out of memory?
[2024-06-14 19:17:03 +0000] [14] [INFO] Booting worker with pid: 14
```

Bref, une fois sur deux les requêtes ne passent pas. Merci pour le coup de main !

--- a/README.md
+++ b/README.md
@@ -42,6 +42,7 @@ Le scripte peut désormais être configuré en utilisant des variables d'environ
 - `GUNICORN_PORT`: Le port sur lequel le serveur proxy interne est en cours d'exécution. Par défaut, il est défini sur '8080'.
 - `GUNICORN_WORKERS`: Le nombre de travailleurs Gunicorn à utiliser. Par défaut, il est défini sur '4'.
 - `GUNICORN_BINDER`: L'adresse IP sur laquelle le serveur proxy interne est lié. Par défaut, il est défini sur '0.0.0.0'.
+- `GUNICORN_TIMEOUT`: Le délai d'attente pour les requêtes Gunicorn. Par défaut, il est défini sur '120'.
 - `LOG_PATH`: Le chemin du fichier journal pour le serveur proxy. Par défaut, il est défini sur '/app/config/logs/rss-proxy.log'. Il y a une rotaion de fichier journal déja configuré. Attention c'est le chemin dans le container.
 - `LOG_LEVEL`: Le niveau de journalisation pour le serveur proxy. Par défaut, il est défini sur 'INFO'.
 - `DB_PATH`: Le chemin de la base de données SQLite pour le serveur proxy. Par défaut, il est défini sur '/app/config/rss-proxy.db'. Attention c'est le chemin dans le container.

--- a/ygg_rss_proxy/__init__.py
+++ b/ygg_rss_proxy/__init__.py
@@ -8,6 +8,7 @@ options = {
     "bind": f"{settings.gunicorn_binder}:{settings.gunicorn_port}",
     "workers": settings.gunicorn_workers,
     "preload_app": True,
+    "timeout": settings.gunicorn_timeout,
 }
 
 logger.info("----------------------------------------------------------")

--- a/ygg_rss_proxy/logging_config.py
+++ b/ygg_rss_proxy/logging_config.py
@@ -7,10 +7,11 @@ import inspect
 import re
 import stackprinter
 
+
 class SecretFilter:
     def __init__(self, patterns):
         self._patterns = patterns
-        
+
     def __call__(self, record):
         record["message"] = self.redact(record["message"])
         if "stack" in record["extra"]:
@@ -19,47 +20,63 @@ class SecretFilter:
 
     def redact(self, message):
         for pattern in self._patterns:
-            message = re.sub(pattern, "********", message)
+            message = re.sub(pattern, "**<REDACTED>**", message)
         return message
 
+
 patterns = [
-    r'passkey=[^&\s]+',
+    r"passkey=[^&\s]+",
     r"'value': '([^']+)'",
-    r'value=\'[^\']+\'',
-    r'cf_clearance=[^;\s]+',
-    r'ygg_=[^;\s]+'
+    r"value=\'[^\']+\'",
+    r"cf_clearance=[^;\s]+",
+    r"ygg_=[^;\s]+",
 ]
 
-# Configuration de Loguru
 logger.remove()
 
-# Fonction de formatage des exceptions avec Stackprinter
+
 def format(record):
     format_ = "{time} {level} {function} {message}\n"
+    pats = [
+        r"passkey=[^&\s]+",
+        r"'value': '([^']+)'",
+        r"value=\'[^\']+\'",
+        r"cf_clearance=[^;\s]+",
+        r"ygg_=[^;\s]+",
+        r"[A-Za-z0-9]+\' \[GET\] of ygg_rss_proxy\.app>",
+    ]
+
     if record["exception"] is not None:
-        record["extra"]["stack"] = stackprinter.format(record["exception"], suppressed_vars=[r".*ygg_playload.*", r".*query_params.*"])
+        stack = stackprinter.format(
+            record["exception"],
+            suppressed_vars=[r".*ygg_playload.*", r".*query_params.*"],
+        )
+        for pat in pats:
+            stack = re.sub(pat, "**<REDACTED>**", stack)
+        record["extra"]["stack"] = stack
         format_ += "{extra[stack]}\n"
     return format_
 
-# Ajout des handlers avec le format personnalisé
+
 logger.add(
     sys.stdout,
     format=format,
     level=settings.log_level.value,
     colorize=True,
-    filter=SecretFilter(patterns)
+    filter=SecretFilter(patterns),
 )
 
 logger.add(
     settings.log_path,
     format=format,
     level="DEBUG",
-    rotation="5 MB",
+    rotation="2 MB",
     retention="5 days",
     compression="zip",
     enqueue=True,
-    filter=SecretFilter(patterns)
+    filter=SecretFilter(patterns),
 )
+
 
 class InterceptHandler(logging.Handler):
     def emit(self, record: logging.LogRecord) -> None:
@@ -77,6 +94,7 @@ class InterceptHandler(logging.Handler):
         logger.opt(depth=depth, exception=record.exc_info).log(
             level, record.getMessage()
         )
+
 
 logging.basicConfig(handlers=[InterceptHandler()], level=0, force=True)
 logging.getLogger("flask").setLevel(logging.DEBUG)

--- a/ygg_rss_proxy/settings.py
+++ b/ygg_rss_proxy/settings.py
@@ -36,6 +36,7 @@ class Settings(BaseSettings):
     gunicorn_workers: int = 4
     gunicorn_port: int = 8080
     gunicorn_binder: str = "0.0.0.0"
+    gunicorn_timeout: int = 120
 
     # LOGGING
     log_level: LogLevel = LogLevel.INFO

